From cfb533287cf6cc0d79369065807c4675f8d41e52 Mon Sep 17 00:00:00 2001
From: Patrick Gaskin <patrick@pgaskin.net>
Date: Wed, 17 Apr 2024 19:21:47 -0400
Subject: [PATCH 1/7] Rename package

So it's easier to test alongside the original one, and so the
original one can be used if needed for SSO.

To update, search for com.realvnc.viewer.android, update where
required. Mostly just provider authorities, content URI names, paths,
and intents. The following command should find all of them. Do not
replace android:name where it refers   to a class (e.g., replace
provider android:authorities but not android:name, replace permission
android:name, don't replace activity/application/receiver/service
android:name).

ag '([^L</\n]|"[^"\n]*/)com[./]realvnc[./]viewer[./]android' | grep -Fve 'cannot be cast to non-null' -e '<fragment ' -e '<androidx' -e '<activity ' -e '<receiver ' -e '<service ' -e '<application ' -e 'apkFileName: ' | grep -E '([^/]|"[^"]*/)com.realvnc.viewer.android[^"]+.'

Test it by ensuring shortcuts and app functionality works, and that the
original one can be installed alongside it.

diff --git a/AndroidManifest.xml b/AndroidManifest.xml
index 21af7a9..372d169 100644
--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8" standalone="no"?><manifest xmlns:android="http://schemas.android.com/apk/res/android" android:compileSdkVersion="34" android:compileSdkVersionCodename="14" android:hardwareAccelerated="true" android:installLocation="auto" package="com.realvnc.viewer.android" platformBuildVersionCode="34" platformBuildVersionName="14">
+<?xml version="1.0" encoding="utf-8" standalone="no"?><manifest xmlns:android="http://schemas.android.com/apk/res/android" android:compileSdkVersion="34" android:compileSdkVersionCodename="14" android:hardwareAccelerated="true" android:installLocation="auto" package="com.realvnc.viewer.android.patched" platformBuildVersionCode="34" platformBuildVersionName="14">
     <supports-screens android:anyDensity="true" android:largeScreens="true" android:normalScreens="true" android:resizeable="true" android:smallScreens="true"/>
     <uses-permission android:name="android.permission.INTERNET"/>
     <uses-permission android:name="com.android.vending.CHECK_LICENSE"/>
@@ -19,10 +19,10 @@
         </intent>
     </queries>
     <uses-permission android:name="com.google.android.gms.permission.AD_ID"/>
-    <permission android:name="com.realvnc.viewer.android.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION" android:protectionLevel="signature"/>
-    <uses-permission android:name="com.realvnc.viewer.android.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION"/>
+    <permission android:name="com.realvnc.viewer.android.patched.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION" android:protectionLevel="signature"/>
+    <uses-permission android:name="com.realvnc.viewer.android.patched.DYNAMIC_RECEIVER_NOT_EXPORTED_PERMISSION"/>
     <application android:allowBackup="false" android:appComponentFactory="androidx.core.app.CoreComponentFactory" android:dataExtractionRules="@xml/data_extraction_rules" android:extractNativeLibs="true" android:fullBackupContent="@xml/backupscheme" android:icon="@mipmap/ic_launcher" android:label="@string/app_name" android:name="com.realvnc.viewer.android.app.Application" android:resizeableActivity="false" android:uiOptions="none">
-        <provider android:authorities="com.realvnc.viewer.android.data.AddressBook" android:exported="false" android:name="com.realvnc.viewer.android.model.AddressBookProvider"/>
+        <provider android:authorities="com.realvnc.viewer.android.patched.data.AddressBook" android:exported="false" android:name="com.realvnc.viewer.android.model.AddressBookProvider"/>
         <meta-data android:name="com.google.android.gms.ads.APPLICATION_ID" android:value="ca-app-pub-2734742209554862~2205718440"/>
         <meta-data android:name="android.content.APP_RESTRICTIONS" android:resource="@xml/app_restrictions"/>
         <activity android:alwaysRetainTaskState="true" android:exported="true" android:label="@string/app_name" android:launchMode="singleTop" android:name="com.realvnc.viewer.android.app.ConnectionChooserActivity" android:theme="@style/ThemeViewerFullscreen" android:windowSoftInputMode="adjustResize|stateUnchanged">
@@ -31,7 +31,7 @@
                 <category android:name="android.intent.category.LAUNCHER"/>
             </intent-filter>
             <intent-filter>
-                <action android:name="com.realvnc.viewer.android.CONNECTIONS"/>
+                <action android:name="com.realvnc.viewer.android.patched.CONNECTIONS"/>
                 <category android:name="android.intent.category.DEFAULT"/>
             </intent-filter>
             <intent-filter>
@@ -43,12 +43,12 @@
             <intent-filter>
                 <action android:name="android.intent.action.VIEW"/>
                 <category android:name="android.intent.category.DEFAULT"/>
-                <data android:scheme="com.realvnc.viewer.android"/>
+                <data android:scheme="com.realvnc.viewer.android.patched"/>
             </intent-filter>
         </activity>
         <activity android:exported="true" android:launchMode="singleInstance" android:name="com.realvnc.viewer.android.app.DesktopActivity" android:taskAffinity="" android:theme="@style/ThemeViewerFullscreen" android:windowSoftInputMode="adjustResize|stateUnchanged">
             <intent-filter>
-                <action android:name="com.realvnc.viewer.android.CONNECT"/>
+                <action android:name="com.realvnc.viewer.android.patched.CONNECT"/>
                 <category android:name="android.intent.category.DEFAULT"/>
                 <data android:mimeType="vnd.android.cursor.item/vnd.realvnc.entry"/>
             </intent-filter>
@@ -59,7 +59,7 @@
                 <category android:name="android.intent.category.DEFAULT"/>
             </intent-filter>
         </activity>
-        <provider android:authorities="com.realvnc.viewer.android.provider" android:exported="false" android:grantUriPermissions="true" android:name="androidx.core.content.FileProvider">
+        <provider android:authorities="com.realvnc.viewer.android.patched.provider" android:exported="false" android:grantUriPermissions="true" android:name="androidx.core.content.FileProvider">
             <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/provider_paths"/>
         </provider>
         <service android:exported="false" android:name="com.realvnc.viewer.android.app.ConnectionService" android:stopWithTask="true"/>
@@ -69,7 +69,7 @@
             </intent-filter>
         </receiver>
         <meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version"/>
-        <provider android:authorities="com.realvnc.viewer.android.androidx-startup" android:exported="false" android:name="androidx.startup.InitializationProvider">
+        <provider android:authorities="com.realvnc.viewer.android.patched.androidx-startup" android:exported="false" android:name="androidx.startup.InitializationProvider">
             <meta-data android:name="androidx.emoji2.text.EmojiCompatInitializer" android:value="androidx.startup"/>
             <meta-data android:name="androidx.lifecycle.ProcessLifecycleInitializer" android:value="androidx.startup"/>
             <meta-data android:name="androidx.profileinstaller.ProfileInstallerInitializer" android:value="androidx.startup"/>
diff --git a/smali/androidx/core/content/FileProvider.smali b/smali/androidx/core/content/FileProvider.smali
index 9e9a067..fef8a20 100644
--- a/smali/androidx/core/content/FileProvider.smali
+++ b/smali/androidx/core/content/FileProvider.smali
@@ -168,7 +168,7 @@
 .method public static c(Landroid/content/Context;Ljava/io/File;)Landroid/net/Uri;
     .locals 1
 
-    const-string v0, "com.realvnc.viewer.android.provider"
+    const-string v0, "com.realvnc.viewer.android.patched.provider"
 
     invoke-static {p0, v0}, Landroidx/core/content/FileProvider;->b(Landroid/content/Context;Ljava/lang/String;)Landroidx/core/content/h;
 
diff --git a/smali/com/realvnc/viewer/android/app/Application.smali b/smali/com/realvnc/viewer/android/app/Application.smali
index bf3e4d2..d1929e7 100644
--- a/smali/com/realvnc/viewer/android/app/Application.smali
+++ b/smali/com/realvnc/viewer/android/app/Application.smali
@@ -35,7 +35,7 @@
 
     new-instance v0, Landroid/content/Intent;
 
-    const-string v1, "com.realvnc.viewer.android.CONNECT"
+    const-string v1, "com.realvnc.viewer.android.patched.CONNECT"
 
     invoke-direct {v0, v1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
 
@@ -337,7 +337,7 @@
 
     move-result-object v1
 
-    const-string v2, "com.realvnc.viewer.android.model.LegacyAddressBookUpgraderTest"
+    const-string v2, "com.realvnc.viewer.android.patched.model.LegacyAddressBookUpgraderTest"
 
     invoke-virtual {v1, v2}, Ljava/lang/ClassLoader;->loadClass(Ljava/lang/String;)Ljava/lang/Class;
     :try_end_0
@@ -395,7 +395,7 @@
 
     invoke-virtual {v4, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
-    const-string v1, "/com.realvnc.viewer.android"
+    const-string v1, "/com.realvnc.viewer.android.patched"
 
     invoke-virtual {v4, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
diff --git a/smali/com/realvnc/viewer/android/app/ConnectionChooserActivity.smali b/smali/com/realvnc/viewer/android/app/ConnectionChooserActivity.smali
index 9845171..6769962 100644
--- a/smali/com/realvnc/viewer/android/app/ConnectionChooserActivity.smali
+++ b/smali/com/realvnc/viewer/android/app/ConnectionChooserActivity.smali
@@ -3625,7 +3625,7 @@
     invoke-static {v0}, Lw2/j0;->E(Ljava/lang/String;)V
 
     :cond_2
-    const-string v0, "com.realvnc.viewer.android://signin"
+    const-string v0, "com.realvnc.viewer.android.patched://signin"
 
     invoke-virtual {p1}, Landroid/content/Intent;->getDataString()Ljava/lang/String;
 
@@ -3703,7 +3703,7 @@
     goto :goto_2
 
     :cond_5
-    const-string v8, "com.realvnc.viewer.android.CONNECT"
+    const-string v8, "com.realvnc.viewer.android.patched.CONNECT"
 
     invoke-virtual {v7, v8}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
 
diff --git a/smali/com/realvnc/viewer/android/app/ConnectionService.smali b/smali/com/realvnc/viewer/android/app/ConnectionService.smali
index 0037714..7480cd9 100644
--- a/smali/com/realvnc/viewer/android/app/ConnectionService.smali
+++ b/smali/com/realvnc/viewer/android/app/ConnectionService.smali
@@ -316,7 +316,7 @@
 
     invoke-direct {v3, p0, v4}, Landroid/content/Intent;-><init>(Landroid/content/Context;Ljava/lang/Class;)V
 
-    const-string v4, "com.realvnc.viewer.android.ACTION_FINISH"
+    const-string v4, "com.realvnc.viewer.android.patched.ACTION_FINISH"
 
     invoke-virtual {v3, v4}, Landroid/content/Intent;->setAction(Ljava/lang/String;)Landroid/content/Intent;
 
@@ -1356,7 +1356,7 @@
 
     move-result-object p1
 
-    const-string p2, "com.realvnc.viewer.android.ACTION_FINISH"
+    const-string p2, "com.realvnc.viewer.android.patched.ACTION_FINISH"
 
     invoke-static {p1, p2}, Landroid/text/TextUtils;->equals(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Z
 
diff --git a/smali/com/realvnc/viewer/android/app/DesktopActivity.smali b/smali/com/realvnc/viewer/android/app/DesktopActivity.smali
index 9e8487e..792209f 100644
--- a/smali/com/realvnc/viewer/android/app/DesktopActivity.smali
+++ b/smali/com/realvnc/viewer/android/app/DesktopActivity.smali
@@ -4394,7 +4394,7 @@
 
     move-result-object v3
 
-    const-string v4, "com.realvnc.viewer.android.CONNECT"
+    const-string v4, "com.realvnc.viewer.android.patched.CONNECT"
 
     invoke-virtual {p1}, Landroid/content/Intent;->getAction()Ljava/lang/String;
 
@@ -4437,7 +4437,7 @@
     goto :goto_2
 
     :cond_2
-    const-string v0, "com.realvnc.viewer.android.ACTION_FINISH"
+    const-string v0, "com.realvnc.viewer.android.patched.ACTION_FINISH"
 
     invoke-virtual {p1}, Landroid/content/Intent;->getAction()Ljava/lang/String;
 
diff --git a/smali/com/realvnc/viewer/android/model/AddressBookProvider.smali b/smali/com/realvnc/viewer/android/model/AddressBookProvider.smali
index e7cf042..97c3e8a 100644
--- a/smali/com/realvnc/viewer/android/model/AddressBookProvider.smali
+++ b/smali/com/realvnc/viewer/android/model/AddressBookProvider.smali
@@ -23,7 +23,7 @@
 
     const/4 v2, 0x1
 
-    const-string v3, "com.realvnc.viewer.android.data.AddressBook"
+    const-string v3, "com.realvnc.viewer.android.patched.data.AddressBook"
 
     invoke-virtual {v0, v3, v1, v2}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
 
diff --git a/smali/w2/e.smali b/smali/w2/e.smali
index 11b9053..4c07ece 100644
--- a/smali/w2/e.smali
+++ b/smali/w2/e.smali
@@ -93,7 +93,7 @@
 
     const/4 v2, 0x1
 
-    const-string v3, "com.realvnc.viewer.android.data.AddressBook"
+    const-string v3, "com.realvnc.viewer.android.patched.data.AddressBook"
 
     invoke-virtual {v0, v3, v1, v2}, Landroid/content/UriMatcher;->addURI(Ljava/lang/String;Ljava/lang/String;I)V
 
@@ -179,7 +179,7 @@
 
     new-instance v0, Ljava/lang/StringBuilder;
 
-    const-string v1, "content://com.realvnc.viewer.android.data.AddressBook/entries/"
+    const-string v1, "content://com.realvnc.viewer.android.patched.data.AddressBook/entries/"
 
     invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
 
diff --git a/smali/w2/f1.smali b/smali/w2/f1.smali
index a1ab8ee..f6ed3fe 100644
--- a/smali/w2/f1.smali
+++ b/smali/w2/f1.smali
@@ -74,9 +74,9 @@
     return-void
 
     :pswitch_1
-    const-string v0, "/com.realvnc.viewer.android"
+    const-string v0, "/com.realvnc.viewer.android.patched"
 
-    const-string v2, "/com.realvnc.viewer.android/.vnc/policy.d"
+    const-string v2, "/com.realvnc.viewer.android.patched/.vnc/policy.d"
 
     invoke-static {}, Landroid/os/Looper;->prepare()V
 
@@ -112,7 +112,7 @@
 
     sget v4, Lcom/realvnc/viewer/android/app/Application;->g:I
 
-    const-string v5, "com.realvnc.viewer.android"
+    const-string v5, "com.realvnc.viewer.android.patched"
 
     const/4 v4, 0x0
 
diff --git a/smali/w2/k1.smali b/smali/w2/k1.smali
index d3695d8..56ca9f0 100644
--- a/smali/w2/k1.smali
+++ b/smali/w2/k1.smali
@@ -17,7 +17,7 @@
 
     invoke-direct {v0, v1}, Landroid/content/UriMatcher;-><init>(I)V
 
-    const-string v1, "com.realvnc.viewer.android.data.AddressBook"
+    const-string v1, "com.realvnc.viewer.android.patched.data.AddressBook"
 
     const-string v2, "entries"
 
diff --git a/smali/w2/u0.smali b/smali/w2/u0.smali
index 1ee2415..64a9be7 100644
--- a/smali/w2/u0.smali
+++ b/smali/w2/u0.smali
@@ -31,7 +31,7 @@
 
     sget-object v2, Landroidx/security/crypto/EncryptedSharedPreferences$PrefValueEncryptionScheme;->AES256_GCM:Landroidx/security/crypto/EncryptedSharedPreferences$PrefValueEncryptionScheme;
 
-    const-string v3, "com.realvnc.viewer.android_enc_preferences"
+    const-string v3, "com.realvnc.viewer.android.patched.android_enc_preferences"
 
     invoke-static {v3, v0, p1, v1, v2}, Landroidx/security/crypto/EncryptedSharedPreferences;->create(Ljava/lang/String;Ljava/lang/String;Landroid/content/Context;Landroidx/security/crypto/EncryptedSharedPreferences$PrefKeyEncryptionScheme;Landroidx/security/crypto/EncryptedSharedPreferences$PrefValueEncryptionScheme;)Landroid/content/SharedPreferences;
 
